telemetry/event_counters.go

```diff
@@ -19,11 +41,63 @@ func EventSuccessCounter(
 	value := getValue()
 
 	telemetry.IncrCounterWithLabels(
-		[]string{eventSuccessKey},
+		[]string{eventTypeMetricKey},
 		value,
 		[]metrics.Label{
 			{Name: "type", Value: eventType},
 			{Name: "is_successful", Value: successResult},
 		},
 	)
 }
+
+// ProofRequirementCounter increments a counter which tracks the number of claims
+// which require proof for the given proof requirement reason (i.e. not required,
+// probabilistic selection, above compute unit threshold).
+func ProofRequirementCounter(
+	reason ProofRequirementReason,
+	getValue func() float32,
+) {
+	value := getValue()
+
+	isRequired := strconv.FormatBool(reason != ProofNotRequired)
+
+	telemetry.IncrCounterWithLabels(
+		[]string{eventTypeMetricKey},
+		value,
+		[]metrics.Label{
+			{Name: "proof_required_reason", Value: reason},
+			{Name: "is_required", Value: isRequired},
+		},
+	)
+}
+
+// ComputeUnitsCounter increments a counter which tracks the number of compute units
+// which are represented by on-chain claims at the given lifecycle state (i.e. claimed,
+// proven, settled).
+func ComputeUnitsCounter(lifecycleStage ClaimProofStage, claim *prooftypes.Claim) {
+	root := (smt.MerkleRoot)(claim.GetRootHash())
+	computeUnitsFloat := float32(root.Sum())
+
+	telemetry.IncrCounterWithLabels(
+		[]string{eventTypeMetricKey},
+		computeUnitsFloat,
+		[]metrics.Label{
+			{Name: "unit", Value: "compute_units"},
```

Comment:
Something's up with these `"uint"` labels, they're not showing up in grafana. :thinking:

Reply:
I figured it out...